x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportCreateAndFollowIndexAction.java

```
@@ -224,19 +211,51 @@ public ClusterState execute(final ClusterState currentState) throws Exception {
-                                listener::onFailure
-                        ));
+                        ccrLicenseChecker.fetchHistoryUUIDs(client, followerIndexMeta, listener::onFailure, historyUUIDs -> {
+                            clusterService.submitStateUpdateTask("", new ClusterStateUpdateTask() {
+                                @Override
+                                public ClusterState execute(ClusterState currentState) throws Exception {
+                                    return recordFollowerShardHistoryUUIDs(currentState, followerIndexMeta.getIndex(), historyUUIDs);
+                                }
+
+                                @Override
+                                public void onFailure(String source, Exception e) {
+                                    listener.onFailure(e);
+                                }
+
+                                @Override
+                                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                                    client.execute(FollowIndexAction.INSTANCE, request.getFollowRequest(), ActionListener.wrap(
+                                        r -> listener.onResponse(new CreateAndFollowIndexAction.Response(true, true, r.isAcknowledged())),
+                                        e -> {
+                                            logger.error("Failed to initiate the shard following", e);
+                                            listener.onResponse(new CreateAndFollowIndexAction.Response(true, true, false));
+                                        }
+                                    ));
+                                }
+                            });
+                        });
                     } else {
                         listener.onResponse(new CreateAndFollowIndexAction.Response(true, false, false));
```

I wondering if we should throw an exception here. Normally when something isn't acked, if you wait it will go through (or at has a chance of doing it). This one will just get stuck.